src/store/actions/api.actions.js

```diff
@@ -256,19 +256,38 @@ export const setInitialEndpoint = endpointSchema => ({
     payload: endpointSchema,
 });
 
-export const fetchEndpointSchema = flag => (dispatch) => {
+export const fetchEndpointSchema = flag => async (dispatch) => {
     dispatch(getEndpointSchemaRequest());
 
-  // return client.get(`apis${pathname}`) // @TODO: RESTORE when endpoint will be ready
-  //   .then((response) => {
-  //     dispatch(getApiSchemaSuccess(response.data));
-  //   })
-  //   .catch((e) => {
-  //     console.log('FETCH_API_SCHEMA_ERROR', 'Infernal server error', e);
-  //   });
+    try {
+        // Get all server names
+        const response = await client.get('/oauth/servers');
+        const serverNames = await response.data.reduce((acc, item) => {
+            acc.push(item.name);
+
+            return acc;
+        }, []);
+        const pluginsFromApiSchemaWithUpdatedOAuthPlugin = await endpointSchema.plugins.map(item => {
```

Comment:
remove the extra `await` no async io being executed. Also you can combine filter and map to avoid checking inside of the map:
```js
const pluginsFromApiSchemaWithUpdatedOAuthPlugin = endpointSchema.plugins
.filter(item => item.name === 'oauth2')
.map(item => {
  const lens = R.lensPath(['config', 'server_names']);
  return R.set(lens, serverNames, item);
});
 ```

Reply:
God damn `true` **!!!**

Reply:
But in case of `.filter` & `.map` we will receive not all the plugins.

But I prepared something for you, @rafaeljesus 😄